src/mbed_tools/devices/_internal/file_parser.py

```diff
@@ -112,43 +148,50 @@ def read_online_id(file_contents: str) -> Optional[OnlineId]:
     return None
 
 
-def extract_product_code_from_htm(all_files_contents: Iterable[str]) -> Optional[str]:
+def _extract_product_code_from_htm(all_files_contents: Iterable[str]) -> Optional[str]:
     """Return first product code found in files contents, None if not found."""
     for contents in all_files_contents:
-        product_code = read_product_code(contents)
+        product_code = _read_product_code(contents)
         if product_code:
             return product_code
     return None
 
 
-def extract_online_id_from_htm(all_files_contents: Iterable[str]) -> Optional[OnlineId]:
+def _extract_online_id_from_htm(all_files_contents: Iterable[str]) -> Optional[OnlineId]:
     """Return first online id found in files contents, None if not found."""
     for contents in all_files_contents:
-        online_id = read_online_id(contents)
+        online_id = _read_online_id(contents)
         if online_id:
             return online_id
     return None
 
 
-def get_all_htm_files_contents(directories: Iterable[pathlib.Path]) -> List[str]:
-    """Yields all htm files contents found in the list of given directories."""
-    files_in_each_directory = (directory.iterdir() for directory in directories)
-    all_files = itertools.chain.from_iterable(files_in_each_directory)
-    return _read_htm_file_contents(all_files)
+def _get_device_file_paths(directories: Iterable[pathlib.Path]) -> List[pathlib.Path]:
+    return [path for directory in directories for path in directory.iterdir() if _is_valid_file(path)]
+
+
+def _try_read_file_text(file_path: pathlib.Path) -> Optional[str]:
+    try:
+        return file_path.read_text()
+    except OSError:
+        logger.warning(f"The file '{file_path}' could not be read from the device, target may not be identified.")
+        return None
 
 
 def _read_htm_file_contents(all_files: Iterable[pathlib.Path]) -> List[str]:
     htm_files_contents = []
     for file in all_files:
-        if _is_htm_file(file):
-            try:
-                htm_files_contents.append(file.read_text())
-            except OSError:
-                logger.warning(f"The file '{file}' could not be read from the device, target may not be identified.")
+        if _is_mbed_htm(file):
+            contents = _try_read_file_text(file)
+            if contents:
+                htm_files_contents.append(contents)
     return htm_files_contents
 
 
-def _is_htm_file(file: pathlib.Path) -> bool:
-    """Checks whether the file looks like an Mbed HTM file."""
-    extensions = [".htm", ".HTM"]
-    return file.suffix in extensions and not file.name.startswith(".")
+def _is_valid_file(file: pathlib.Path) -> bool:
+    """Checks whether the file is not hidden."""
+    return not file.name.startswith(".")
+
+
+def _is_mbed_htm(path: pathlib.Path) -> bool:
+    return path.suffix.lower() == ".htm"
```

Comment:
Is "index.htm" an mbed_htm?

Reply:
I don't know. I don't have a device with an "index.htm", but the logic of this check hasn't changed in this PR, apart from normalizing the string case so we don't have to check "HTM" and "htm". What are the contents of "index.htm"?

Reply:
I've changed the name to `_is_htm_file` as I'm assuming index.htm is in the set of html files we care about, which could contain the product code.